mirar/pipelines/winter/generator.py

```diff
@@ -255,22 +257,26 @@ def winter_reference_generator(image: Image):
     subdetid = int(image[SUB_ID_KEY])
     logger.debug(f"Fieldid: {fieldid}, subdetid: {subdetid}")
 
-    db_results = RefStack.sql_model().select_query(
-        select_keys=["savepath"],
-        compare_keys=["fieldid", SUB_ID_KEY.lower()],
-        compare_values=[fieldid, subdetid],
-        comparators=["__eq__", "__eq__"],
+    constraints = DBQueryConstraints(
+        columns=["fieldid", SUB_ID_KEY.lower()],
+        accepted_values=[fieldid, subdetid],
+    )
+
+    db_results = select_from_table(
+        db_constraints=constraints,
+        sql_table=RefStack.sql_model,
+        output_columns=["savepath"],
     )
 
     ref_exists = False
     if len(db_results) > 0:
         savepaths = [x[0] for x in db_results]
```

Comment:
savepaths is no longer required?

Reply:
I am confused by this. I'm only renaming the code that you wrote. Is the savepath not needed for `RefFromPath`? You tell me, I may have misunderstood that block!